ext/datastore/bigquery/adapter.go

```diff
@@ -159,6 +226,14 @@ func bqCreateTableMetaAdapter(t BQTable) (meta *bqapi.TableMetadata, err error)
 			meta.RangePartitioning = bqPartitioningRangeTo(*t.Metadata.Partition)
 		}
 	}
+
+	if t.Metadata.Source != nil {
+		meta.ExternalDataConfig, err = bqExternalDataConfigTo(*t.Metadata.Source)
```

Comment:
So how is the table schema gets populated in this case? Is the user going to pass schema just like a normal table?

Reply:
Yes, there are two ways on defining schema for external tables:
- Use autodetect schema It will disable the schema provided  and BQ will attempt to detect the schema by the sheets
- Use user defined schema, BQ will map the google sheets column to the defined schema
On this PR, we detect schema based on user passed schema. For the next development. we can implement the autodetect schema by having the `AutoDetect` option on the `ExternalDataConfig` as true if no schema is supplied by the user.